flink-runtime/src/main/java/org/apache/flink/runtime/security/JaasConfiguration.java

```diff
@@ -0,0 +1,158 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.security;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.Preconditions;
+import org.apache.hadoop.security.authentication.util.KerberosUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
+ * JAAS configuration provider object that provides default LoginModule for various connectors that supports
+ * JAAS/SASL based Kerberos authentication. The implementation is inspired from Hadoop UGI class.
+ *
+ * Different connectors uses different login module name to implement JAAS based authentication support.
+ * For example, Kafka expects the login module name to be "kafkaClient" whereas ZooKeeper expect the
+ * name to be "client". This sets onus on the Flink cluster administrator to configure/provide right
+ * JAAS config entries. To simplify this requirement, we have introduced this abstraction that provides
+ * a standard lookup to get the login module entry for the JAAS based authentication to work.
+ *
+ * HDFS connector will not be impacted with this configuration since it uses UGI based mechanism to authenticate.
+ *
+ * <a href="https://docs.oracle.com/javase/7/docs/api/javax/security/auth/login/Configuration.html">Configuration</a>
+ *
+ */
+
+@Internal
+public class JaasConfiguration extends Configuration {
+
+	private static final Logger LOG = LoggerFactory.getLogger(JaasConfiguration.class);
+
+	public static final String JAVA_VENDOR_NAME = System.getProperty("java.vendor");
+
+	public static final boolean IBM_JAVA;
+
+	static {
+		IBM_JAVA = JAVA_VENDOR_NAME.contains("IBM");
+	}
+
+	public JaasConfiguration(String keytab, String principal) {
+
+		LOG.info("Initializing JAAS configuration instance. Parameters: {}, {}", keytab, principal);
+
+		if(!Preconditions.isNullOrEmpty(keytab) && !Preconditions.isNullOrEmpty(principal)) {
+
+			if(IBM_JAVA) {
+				keytabKerberosOptions.put("useKeytab", prependFileUri(keytab));
+				keytabKerberosOptions.put("credsType", "both");
+			} else {
+				keytabKerberosOptions.put("keyTab", keytab);
+				keytabKerberosOptions.put("doNotPrompt", "true");
+				keytabKerberosOptions.put("useKeyTab", "true");
+				keytabKerberosOptions.put("storeKey", "true");
+			}
+
+			keytabKerberosOptions.put("principal", principal);
+			keytabKerberosOptions.put("refreshKrb5Config", "true");
+			keytabKerberosOptions.putAll(debugOptions);
+
+			keytabKerberosAce = new AppConfigurationEntry(
+					KerberosUtil.getKrb5LoginModuleName(),
+					AppConfigurationEntry.LoginModuleControlFlag.REQUIRED,
+					keytabKerberosOptions);
+		}
+	}
+
+	private static final Map<String, String> debugOptions = new HashMap<>();
```

Comment:
Further, this static field should be removed before we merge. Please use logging and control the logging via log4j or logback configuration.